holoviews/plotting/bokeh/raster.py

```diff
@@ -156,11 +158,24 @@ def get_data(self, element, ranges, style):
             X, Y = colormesh(X, Y)
             zvals = zdata.T.flatten() if self.invert_axes else zdata.flatten()
             XS, YS = [], []
-            for x, y, zval in zip(X, Y, zvals):
-                if np.isfinite(zval):
-                    XS.append(list(x[:-1]))
-                    YS.append(list(y[:-1]))
-            data = {'xs': XS, 'ys': YS, z.name: zvals[np.isfinite(zvals)]}
+            mask = []
+            xc, yc = [], []
+            for xs, ys, zval in zip(X, Y, zvals):
+                xs, ys = xs[:-1], ys[:-1]
+                if np.isfinite(zval) and all(np.isfinite(c) for c in xs) and all(np.isfinite(c) for c in ys):
+                    XS.append(list(xs))
+                    YS.append(list(ys))
+                    mask.append(True)
+                    if 'hover' in self.handles:
+                        xc.append(xs.mean())
+                        yc.append(ys.mean())
+                else:
+                    mask.append(False)
+
+            data = {'xs': XS, 'ys': YS, z.name: zvals[np.array(mask)]}
```

Comment:
Maybe I am misunderstanding, but why can't you just compute the mask and apply that to ``xs`` and ``ys`` instead of defining ``XS`` and ``YS``?

Reply:
XS and YS are lists of arrays and not arrays themselves.

Reply:
To be perfectly accurate they are lists of lists since the overhead for serializing and deserializing a bunch of tiny arrays is much larger than any benefit gained.